recipes/openssl/3.x.x/conanfile.py

```diff
@@ -551,8 +569,8 @@ def package(self):
                 if file.endswith(".a"):
                     os.unlink(os.path.join(libdir, file))
 
-        if not self.options.no_fips:
-            provdir = os.path.join(self.source_folder, "providers")
+        if self._is_fips_enabled:
+            provdir = self._fips_provider_dir
```

Comment:
Works as expected 👍🏼 
Just a minor thing, does it help to add a `self.output.info()` to print the `provdir` path ?
It will make it easier to confirm that the certified fips module has been copied from 3.0.9 directory.

Reply:
Good point @kulkarniamit, I'll see what I can do... 

Actually, ideally, I was trying to add some testing logic in the `test_package` code to be able to properly verify the version of the `fips` module... For example, that `version == 3.0.9` when `use_validated_fips == True` or `version == tested_package_version` otherwwise, but both programmatically or simply using the `openssl list -providers` command require a properly `fipsinstall`ed module before loading it to check the version...

Do you know of any ways to check the version of the `fips` module without it needing to be fully installed?

Reply:
As you mentioned, printing buildinfo related information for fips provider requires loading the provider. Loading FIPS provider requires having a configuration file with `module-mac` to enable self tests of the module.

If we don't prefer checking the integration with `openssl` command, we are left with two options:
* Use `strings` command (or any other equivalent) and search for presence of version number in module. Need to ensure we've a consistent command or python string search that platform independent
  ```bash
  # Linux
  $ strings <path_to_fips_module> | grep -i -A1 buildinfo
  buildinfo
  3.0.9
  ```
* Print the fips module source and destination directory information

Though its extra work, running `openssl fipsinstall` to generate `fipsmodule.cnf` and loading the provider programmatically in `test_package.c` would be the ideal future-proof platform-independent approach to printing version information.